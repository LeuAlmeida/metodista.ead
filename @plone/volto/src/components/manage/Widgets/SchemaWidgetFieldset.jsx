/**
 * Schema widget fieldset.
 * @module components/manage/Widgets/SchemaWidgetFieldset
 */

import React from 'react';
import PropTypes from 'prop-types';
import { DragSource, DropTarget } from 'react-dnd';
import { Icon } from 'semantic-ui-react';

/**
 * Schema widget fieldset component.
 * @function SchemaWidgetFieldset
 * @returns {string} Markup of the component.
 */
export const SchemaWidgetFieldsetComponent = ({
  connectDragSource,
  connectDragPreview,
  connectDropTarget,
  isDragging,
  title,
  order,
  active,
  onShowEditFieldset,
  onShowDeleteFieldset,
  onClick,
}) =>
  connectDropTarget(
    connectDragPreview(
      <div
        className={`item${active ? ' active' : ''}`}
        style={{ opacity: isDragging ? 0.5 : 1 }}
        onClick={() => onClick(order)}
      >
        {connectDragSource(
          <i aria-hidden="true" className="grey bars icon drag handle" />,
        )}
        {title}
        <a
          onClick={event => {
            event.stopPropagation();
            onShowEditFieldset(order);
          }}
        >
          <Icon name="write square" size="large" color="blue" />
        </a>
        <a
          onClick={event => {
            event.stopPropagation();
            onShowDeleteFieldset(order);
          }}
        >
          <Icon name="close" size="large" color="red" />
        </a>
      </div>,
    ),
  );

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
SchemaWidgetFieldsetComponent.propTypes = {
  connectDragPreview: PropTypes.func.isRequired,
  connectDragSource: PropTypes.func.isRequired,
  connectDropTarget: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired,
  order: PropTypes.number.isRequired,
  active: PropTypes.bool.isRequired,
  onOrderFieldset: PropTypes.func.isRequired,
  onShowEditFieldset: PropTypes.func.isRequired,
  onShowDeleteFieldset: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default DropTarget(
  'fieldset',
  {
    hover(props, monitor) {
      const dragOrder = monitor.getItem().order;
      const hoverOrder = props.order;

      if (dragOrder === hoverOrder) {
        return;
      }

      props.onOrderFieldset(dragOrder, hoverOrder - dragOrder);

      monitor.getItem().order = hoverOrder;
    },
  },
  connect => ({
    connectDropTarget: connect.dropTarget(),
  }),
)(
  DragSource(
    'fieldset',
    {
      beginDrag(props) {
        return {
          id: props.label,
          order: props.order,
        };
      },
    },
    (connect, monitor) => ({
      connectDragSource: connect.dragSource(),
      connectDragPreview: connect.dragPreview(),
      isDragging: monitor.isDragging(),
    }),
  )(SchemaWidgetFieldsetComponent),
);
