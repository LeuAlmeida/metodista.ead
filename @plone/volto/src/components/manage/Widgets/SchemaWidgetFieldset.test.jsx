import React from 'react';
import renderer from 'react-test-renderer';

import { SchemaWidgetFieldsetComponent } from './SchemaWidgetFieldset';

test('renders a contents item component', () => {
  const component = renderer.create(
    <SchemaWidgetFieldsetComponent
      connectDragSource={x => x}
      connectDragPreview={x => x}
      connectDropTarget={x => x}
      order={1}
      isDragging={false}
      active={false}
      onOrderFieldset={x => x}
      onShowEditFieldset={x => x}
      onShowDeleteFieldset={x => x}
      onClick={x => x}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
