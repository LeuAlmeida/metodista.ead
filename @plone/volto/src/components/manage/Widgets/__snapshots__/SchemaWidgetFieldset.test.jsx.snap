// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a contents item component 1`] = `
<div
  className="item"
  onClick={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <i
    aria-hidden="true"
    className="grey bars icon drag handle"
  />
  <a
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="blue write square large icon"
    />
  </a>
  <a
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="red close large icon"
    />
  </a>
</div>
`;
